var-core/src/main/java/io/varhttp/parameterhandlers/Convert.java

```diff
@@ -7,7 +7,7 @@
 
 public class Convert {
 	public Object convert(String parameter, Class<?> type, String defaultValue) {
-		if (parameter == null) {
+		if (parameter == null || ("".equals(parameter) && !String.class.equals(type))) {
```

Comment:
May be we should bring the later check into a separated method with better name so it is easier to understand what we want to check.